From 040b9249631d1d14d4b184440cbb40485606c06d Mon Sep 17 00:00:00 2001
From: jzhuan5 <jin.can.zhuang@intel.com>
Date: Fri, 24 May 2013 17:40:15 -0400
Subject: [PORT FROM MAIN]adbd: adb root should terminate adbd only if it's debuggable

BZ: 110652

adb root command will terminate adbd when closing the socket if current
user id is not root. This works for userdebug build, as adb root causes
re-enumeration.

But for user build, adb root command won't cause re-enumeration, and if
adbd is terminated and restarted, it will be in offline state, thus it
won't send any thing to host.

Change-Id: Ie2e844221f3474402a7e2a8535f84ee86a631994
Orig-Change-Id: I68825bc79358d87a3e1b41f4658cd484e8eadb1e
Category: aosp improvement
Domain: USB ADB
Origin: internal
Upstream-Candidate: no, proprietary
Signed-off-by: jzhuan5 <jin.can.zhuang@intel.com>
---
 adb/sockets.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/adb/sockets.c b/adb/sockets.c
index 5c1a074..5581cd7 100644
--- a/adb/sockets.c
+++ b/adb/sockets.c
@@ -417,6 +417,7 @@ asocket *create_local_service_socket(const char *name)
     int fd;
 #if !ADB_HOST
     char value[PROPERTY_VALUE_MAX];
+    char debug[PROPERTY_VALUE_MAX];
 #endif
 
 #if !ADB_HOST
@@ -435,7 +436,9 @@ asocket *create_local_service_socket(const char *name)
 
 #if !ADB_HOST
     property_get("service.adb.tcp.port", value, "0");
-    if ((!strncmp(name, "root:", 5) && getuid() != 0)
+    property_get("ro.debuggable", debug, "");
+    if ((!strncmp(name, "root:", 5) && getuid() != 0 &&
+         strcmp(debug, "1") == 0)
         || (!strncmp(name, "usb:", 4) && strcmp(value, "0"))
         || !strncmp(name, "tcpip:", 6)) {
         D("LS(%d): enabling exit_on_close\n", s->id);
-- 
1.7.4.1

